### ISSUE TITLE
getters are executed when using td.constructor()

### ISSUE THREAD

**Issue by kai670** [testdouble/testdouble.js#218]
It seems that getters are executed when using `td.constructor`. The next code:

```javascript
class Foo {
  constructor() {
    this.a = {b: 'b'};
  }
  get bar() {
    return this.a.b;
  }
}
let FakeFoo = td.constructor(Foo);
```

Fails with:
```
Uncaught TypeError: Cannot read property 'b' of undefined
    at Object.get (Foo.test.js:11)
    at testdouble.js?9db23c8…:972
    at arrayFilter (testdouble.js?9db23c8…:1898)
    at Object.filter (testdouble.js?9db23c8…:6636)
    at allPrototypeFunctionNames (testdouble.js?9db23c8…:971)
    at module.exports (testdouble.js?9db23c8…:967)
    at testdouble.js?9db23c8…:123
    at Object.tap (testdouble.js?9db23c8…:8344)
    at fakeConstructorFromType (testdouble.js?9db23c8…:103)
    at new module.exports (testdouble.js?9db23c8…:96)
```

AFAIK this makes impossible replacing classes like this that have getters that depend on values set in the constructor.

**Comment by searls** [comment]
This is a great point! I had not considered this before! (I saw some [tweets](https://twitter.com/armorik83/status/844571456130334720) referencing getters but I didn't look closely.)

I think we should definitely add this to the test suite and figure out what to do.

**Comment by searls** [comment]
Resolved by #254 and landed in testdouble@3.0

This should be fixed now
